package Render;

message MaterialDesc
{
    enum ParameterType
    {
        VECTOR1 = 1;
        VECTOR2 = 2;
        VECTOR3 = 3;
        VECTOR4 = 4;
    }

    enum ParameterSemantic
    {
        GENERIC = 1;
        COLOR = 2;
    }

    message Vector4
    {
        required float x = 1;
        required float y = 2;
        required float z = 3;
        required float w = 4;
    }

    message Parameter
    {
        required string Name = 1;
        required ParameterType Type = 2;
        required ParameterSemantic Semantic = 3;
        required uint32 Register = 4;
        required Vector4 Value = 5;
    }
    
    required string Name = 1;
    required string VertexProgram = 2;
    required string FragmentProgram = 3;
    repeated Parameter VertexParameters = 4;
    repeated Parameter FragmentParameters = 5;
}

